src/components/TopQueries/TopQueries.ts

```diff
@@ -0,0 +1,122 @@
+import {
+    $$,
+    Component,
+    ComponentOptions,
+    IAnalyticsActionCause,
+    IComponentBindings,
+    Initialization,
+    IQuerySuggestRequest,
+    l,
+    IQuerySuggestResponse,
+} from 'coveo-search-ui';
+import './Strings';
+
+export interface ITopQueriesOptions {
+    /**
+     * The parameters sent to the suggestion query.
+     * The component uses this information to get better suggestions
+     */
+    suggestionQueryParams?: IQuerySuggestRequest;
+    /**
+     * The displayed title over the suggestions
+     */
+    title?: string;
+    /**
+     * Specifies the handler called when a suggestion is clicked.
+     * This should call a search query.
+     *
+     * Default navigates to #q=<suggestion>
+     */
+    onClick?: (expression: string) => void;
+}
+
+/**
+ * The TopQueries component suggests the top searched queries in the specific context and links the search results of the suggestions to the user
+ */
+export class TopQueries extends Component {
+    static ID = 'TopQueries';
+
+    /**
+     * The possible options for the TopQueries component
+     * @componentOptions
+     */
+    static options: ITopQueriesOptions = {
+        suggestionQueryParams: ComponentOptions.buildJsonOption<IQuerySuggestRequest>({ defaultValue: { q: '' } }),
+        title: ComponentOptions.buildStringOption({ defaultValue: l('TopQueries_title') }),
+        onClick: ComponentOptions.buildCustomOption((s) => null, {
+            defaultValue: (expression: string) => {
+                window.location.href = `#q=${expression}`;
```

Comment:
Instead of playing directly with the URL maybe we should use the QueryStateManager ? 😸

Reply:
Absolutely!